donut/modules/groups/helpers.py

```diff
@@ -82,3 +82,28 @@ def get_group_data(group_id, fields=None):
     # Return the row in the form of a dict
     result = {f: t for f, t in zip(fields, result)}
     return result
+
+
+
+def get_position_data(fields=None):
+    all_returnable_fields = ["group_id", "pos_id", "pos_name"]
+    default_fields = ["user_id", "pos_id", "pos_name", "group_id"]
+
+    if fields is None:
+        fields = default_fields
+    else:
+        if any(f not in all_returnable_fields for f in fields):
+            return "Invalid field"
+
+    #s= sqlalchemy.sql.select(fields).select_from(sqlalchemy.text("positions JOIN position_holders WHERE positions.pos_id = position_holders.pos_id"))
+    
+    
+    result = flask.g.db.execute("SELECT user_id, positions.pos_id, pos_name, positions.group_id \nFROM positions JOIN position_holders WHERE positions.pos_id = position_holders.pos_id;")    
```

Comment:
NATURAL JOIN the tables.
Use the list of fields to query rather than querying for everything and filtering in python